core/src/main/scala/org/elasticmq/actor/queue/QueueActorMessageOps.scala

```diff
@@ -185,4 +186,27 @@ trait QueueActorMessageOps extends Logging {
       }
     }
   }
+
+  private def moveToDeadLetterQueue(message: InternalMessage): Unit = {
+    def moveMessageToDLQ(internalMessage: InternalMessage): Unit = {
+      messageQueue += internalMessage
+      logger.debug(s"Moved message with id ${internalMessage.id} to DLQ ${queueData.name}")
+    }
+
+    copyMessagesToActorRef.foreach { _ ! SendMessage(message.toNewMessageData) }
+
+    if (queueData.isFifo) {
+      // Ensure a message with the same deduplication id is not on the queue already. If the message is already on the
+      // queue do nothing.
+      // TODO: A message dedup id should be checked up to 5 mins after it has been received. If it has been deleted
```

Comment:
so we'd have to maintain a cache of already received message ids (for the last 5 mins)?

Reply:
Seems so. This comment was already there (in similar function). I have copied it to not forgot about this place once someone will do this TODO. 
I have not extracted those functions to one common as I feel that behaviour is quite similar but the differences between input/output parameters are too big.

Reply:
ok :) We'll get back to these 5 minutes in #354 ;)